### PR TITLE
Add Penetration Testing

### PR DESCRIPTION
**Pull request by meitar**
**https://github.com/enaqx/awesome-pentest#readme**

**A curated selection of penetration testing resources, compiled to aid security researchers and infosec students in finding high-quality tools for discovering, analyzing, and reporting on security vulnerabilities.**

### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

## Requirements for your pull request

- **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).** Try to prioritize unreviewed PRs, but you can also add more comments to reviewed PRs. Go through the below list when reviewing. This requirement is meant to help make the Awesome project self-sustaining. Comment here which PRs you reviewed. You're expected to put a good effort into this and to be thorough. Look at previous PR reviews for inspiration.
	- I have reviewed PRs #1346 and #1364.
- I have read and understood the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/master/create-list.md).
- This pull request has a descriptive title in the format `Add Name of List` (Example: `Add Swift`), not `Update readme.md` or `Add awesome list`.
- The entry in the Awesome list should:
	- Include a short description about the project/theme of the list. **It should not describe the list itself.**<br>Example: `- [Fish](…) - User-friendly shell.`, not `- [Fish](…) - Resources for Fish.`.
	- Be added at the bottom of the appropriate category.
- The list I'm submitting complies with the below requirements.

## Requirements for your Awesome list

- **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
- It's the result of hard work and the best I could possibly produce.
- Non-generated Markdown file in a GitHub repo.
- **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
- The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- Not a duplicate.
- Only has awesome items. Awesome lists are curations of the best, not everything.
- Includes a project logo/illustration whenever possible.
	- Either fullwidth or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- The image should link to the project website or any relevant website.
	- The image should be high-DPI. Set it to maximum half the width of the original image.
- Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- Includes the [Awesome badge](https://github.com/sindresorhus/awesome/blob/master/awesome.md#awesome-badge).
	- Should be placed on the right side of the readme heading.
	- Should link back to this list.
- Has a Table of Contents section.
	- Should be named `Contents`, not `Table of Contents`.
	- Should be the first section in the list.
	- Should only have one level of sub-lists, preferably none.
- Has an [appropriate license](https://github.com/sindresorhus/awesome/blob/master/awesome.md#choose-an-appropriate-license).
	- That means something like CC0, **not a code licence like MIT, BSD, Apache, etc.**
	- [WTFPL](http://www.wtfpl.net) and [Unlicense](http://unlicense.org) are not acceptable licenses.
	- If you use a license badge, it should be SVG, not PNG.
- Has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/awesome.md#include-contribution-guidelines).
	- The file should be named `contributing.md`. Casing is up to you.
- Has consistent formatting and proper spelling/grammar.
	- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- The description starts with an uppercase character and ends with a period.
	- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- Doesn't include a Travis badge.<br>You can still use Travis for list linting, but the badge has no value in the readme.

Go to the top and read it again.

### CODE REVIEW

**Comment by meitar** [comment]
> Fails awesome-lint with multiple errors. Please double check *all* of the guidelines.
> 
> The most frequent error from awesome-lint is: `Marker style should be -`

AFAICT, marker style was not a requirement for inclusion as early as yesterday when this PR was submitted and is not mentioned in the guidelines.

`awesome-lint` also does not seem to have a check for this, until after you proposed sindresorhus/awesome-lint#29 8 hours ago, which is *after* this PR was opened.

If these are new requirements, fine, but I just want to be clear about the timeline of events regarding when it is possible to "double check *all* of the guidelines* and when one must have the ability to time travel to do so. :)

**Comment by transitive-bullshit** [comment]
Hey @meitar, not sure where your comment went, but your feedback is perfectly reasonable. 👍 

Sorry for any confusion; we're trying to normalize the specific linting rules at the moment, and awesome-lint has never officially been a requirement for submission. My feedback here is part of an attempt to feel out how well awesome-lint would work for these purposes and fine-tune the specific set of rules.

Please follow-up with this [issue](https://github.com/sindresorhus/awesome-lint/issues/30) if you have feedback on any of the proposed rules to awesome-lint.

Thanks!

**Comment by meitar** [comment]
@hsluoyz Thank you for noticing this. I've sent a patch upstream.

**Comment by cpuu** [comment]
Some title have an error on mark down depth level

File: readme.md [hunk 1]
```diff
@@ -433,6 +433,7 @@
 - [Vehicle Security and Car Hacking](https://github.com/jaredthecoder/awesome-vehicle-security#readme)
 - [Web Security](https://github.com/qazbnm456/awesome-web-security#readme) - Security of web apps & services.
 - [Lockpicking](https://github.com/meitar/awesome-lockpicking#readme) - The art of unlocking a lock by manipulating its components without the key.
+- [Pentest](https://github.com/enaqx/awesome-pentest#readme) - Penetration testing resources and tools.
```
Comment (sindresorhus) [inline readme.md]: `Pentest` => `Penetration Testing` since that's the title of your list

Comment (sindresorhus) [reply readme.md]: > The entry in the Awesome list should:
Include a short description about the project/theme of the list. **It should not describe the list itself.**
Example: - [Fish](…) - User-friendly shell., not - [Fish](…) - Resources for Fish..

**Comment by sindresorhus** [comment]
```
✖    1:32  Invalid badge source                                                                                             awesome/badge
✖   16:3   ToC item "Open Source Intelligence (OSINT) Resources" does not match corresponding heading "OSINT Resources"     awesome/toc
✖   24:3   ToC item "Network Vulnerability scanners" does not match corresponding heading "Network Vulnerability Scanners"  awesome/toc
✖  342:3   Link to http://hachoir3.readthedocs.io/ is dead                                                                  no-dead-urls
✖  368:3   Link to http://www.ampliasecurity.com/research/windows-credentials-editor/ is dead                               no-dead-urls
✖  372:55  List item description must start with valid casing                                                               awesome/list-item
✖  396:4   Text "OSX" should be written as "macOS"                                                                          awesome/spell-check
✖  438:59  Text "github" should be written as "GitHub"                                                                      awesome/spell-check
✖  457:3   Link to https://github.com/ascribe/image-match] is dead                                                          no-dead-urls
✖  604:3   Link to http://www.wiley.com/WileyCDA/WileyTitle/productCd-0470639539.html is dead                               no-dead-urls
✖  636:3   Link to https://www.0day.today/ is dead                                                                          no-dead-urls
✖  636:38  List item link and description must be separated with a dash                                                     awesome/list-item
✖  636:40  Link to http://mvfjfugdwgc5uwho.onion/ is dead                                                                   no-dead-urls
✖  688:78  List item description must end with proper punctuation                                                           awesome/list-item
✖  727:1   Don’t use multiple top level headings (1:1)                                                                      no-multiple-toplevel-headings
```

**Comment by sindresorhus** [comment]
// @enaqx @techgaun

**Comment by techgaun** [comment]
Thanks @sindresorhus for the heads up. I will clean up sometimes later today

**Comment by sindresorhus** [comment]
Still many things not fixed. @techgaun Do you still intend to do the changes or should I close this?

```
❯ awesome-lint https://github.com/enaqx/awesome-pentest
✖ Linting

  README.md:5:1
  ✖    1:1    License section must be at heading depth 2                                     remark-lint:awesome/license
  ✖    1:1    Exceeded max depth of 2 levels                                                 remark-lint:awesome/toc
  ✖    1:1    Exceeded max depth of 2 levels                                                 remark-lint:awesome/toc
  ✖    1:32   Invalid badge source                                                           remark-lint:awesome/badge
  ✖    5:1    Don’t use emphasis to introduce a section, use a heading                       remark-lint:no-emphasis-as-heading
  ✖   96:3    Link to https://ghostproject.fr/ is dead                                       remark-lint:no-dead-urls
  ✖  136:3    Invalid list item link                                                         remark-lint:awesome/list-item
  ✖  137:3    Invalid list item link                                                         remark-lint:awesome/list-item
  ✖  138:3    Invalid list item link                                                         remark-lint:awesome/list-item
  ✖  139:3    Invalid list item link                                                         remark-lint:awesome/list-item
  ✖  140:3    Invalid list item link                                                         remark-lint:awesome/list-item
  ✖  141:3    Invalid list item link                                                         remark-lint:awesome/list-item
  ✖  142:3    Invalid list item link                                                         remark-lint:awesome/list-item
  ✖  143:3    Invalid list item link                                                         remark-lint:awesome/list-item
  ✖  144:3    Invalid list item link                                                         remark-lint:awesome/list-item
  ✖  145:3    Invalid list item link                                                         remark-lint:awesome/list-item
  ✖  146:3    Invalid list item link                                                         remark-lint:awesome/list-item
  ✖  147:3    Invalid list item link                                                         remark-lint:awesome/list-item
  ✖  148:3    Invalid list item link                                                         remark-lint:awesome/list-item
  ✖  149:3    Invalid list item link                                                         remark-lint:awesome/list-item
  ✖  150:3    Invalid list item link                                                         remark-lint:awesome/list-item
  ✖  151:3    Invalid list item link                                                         remark-lint:awesome/list-item
  ✖  152:3    Invalid list item link                                                         remark-lint:awesome/list-item
  ✖  400:4    Text "OSX" should be written as "macOS"                                        remark-lint:awesome/spell-check
  ✖  505:3    Link to https://samy.pl/poisontap/ is dead                                     remark-lint:no-dead-urls
  ✖  633:3    Link to https://technet.microsoft.com/en-us/security/advisories#APUMA is dead  remark-lint:no-dead-urls
  ✖  641:3    Link to https://www.0day.today/ is dead                                        remark-lint:no-dead-urls
  ✖  641:104  Link to http://mvfjfugdwgc5uwho.onion/ is dead                                 remark-lint:no-dead-urls
  ✖  692:78   List item description must end with proper punctuation                         remark-lint:awesome/list-item
  ✖  732:1    Don’t use multiple top level headings (1:1)                                    remark-lint:no-multiple-toplevel-headings

  30 errors
```

https://github.com/sindresorhus/awesome-lint

**Comment by sindresorhus** [comment]
https://github.com/sindresorhus/awesome/pull/1366/files#r249361717

**Comment by techgaun** [comment]
Sorry I have been pretty busy with life and work. I hope community might jump in to fix the issues you have outlined but I don't think I will be able to get this fixed anytime soon.

**Comment by meitar** [comment]
@sindresorhus @techgaun I can work on some of these issues but a number the linting issues are things I think are problems with the linter. For example, some of the pentest tools have `OSX` in their name, and this is the *correct* name of the tool. We cannot (and should not) change that in our list unless the upstream tool changes its name. Another issue is that some of the links reported as dead are not actually dead, they are just blocking automated tools like the `awesome_bot` Gem from visiting the site (using something like Cloudflare DDoS protection). We have addressed this in our own repo with a customized `awesome_bot` `--white-list`, but they still show up as linting errors here. So I will address what issues I can, but these other issues I will probably just ignore.

**Comment by NewAlexandria** [comment]
I can have a go at linting and fixing this.  I just did it for one of my submissions, and helped a buddy's, too

**Comment by NewAlexandria** [comment]
@meitar and @techgaun what do you guys think about adding links to some review-sites for commercial pentesting services?  I don't want to get into promotion/marketing, but I do think that knowing where to find reputable vendors is acceptable.

I overall like this list.  I'm recently more aware, as I went through a long cycle of reviewing tools and vendor options for pentesting.

**Comment by meitar** [comment]
> @meitar and @techgaun what do you guys think about adding links to some review-sites for commercial pentesting services?

I would be open to discussing this but I think this should be discussed on the awsome-pentest issue tracker, not here.

I'll say briefly that linking to reviews or review sites feels slightly out of scope at first glance because the list is supposed to *already be* a (liberally open but still) "curated" collection of tools and such, but I am open to being convinced otherwise.

**Comment by NewAlexandria** [comment]
moving to the issues, sure.  thx

**Comment by NewAlexandria** [comment]
I've updated this list, and it is now linting with `awesome-lint` @sindresorhus

**Comment by sindresorhus** [comment]
> You have to review at least 2 other open pull requests.

**Comment by sindresorhus** [comment]
You also need to fix the merge conflict here.

**Comment by sindresorhus** [comment]
This is still not fixed: https://github.com/sindresorhus/awesome/pull/1366/files#r249361717

**Comment by meitar** [comment]
> You have to review at least 2 other open pull requests.

I'll be honest: I'm not going to do that.

I was happy to contribute to this list 9 months ago but in the intervening time a number onerous requirements for contributing have been added and I just don't care enough about including an already very well-known list on yet another list in comparison with all the other work I'm currently engaged in elsewhere and care more about. `¯\_(ツ)_/¯`

I'd be happy to pass this PR off to anyone else who wants to take it over. Perhaps @NewAlexandria wants to?

**Pull request closed without merging**